File: app/src/main/java/ru/sukharev/myrecipes/addrecipe/AddRecipeContract.java
package ru.sukharev.myrecipes.addrecipe;

import ru.sukharev.myrecipes.base.BasePresenter;
import ru.sukharev.myrecipes.base.BaseView;

/**
 * Created by pasha on 13.11.17.
 */

public interface AddRecipeContract {

    interface View extends BaseView<Presenter>{

        void onRecipeAdded();

        void onRecipeAddingError();

        void setRating(int rating);
    }

    interface Presenter extends BasePresenter<View> {

        void addRecipe(String title, String desc);

        void setRating(int rating);
    }
}

File: app/src/main/java/ru/sukharev/myrecipes/addrecipe/AddRecipeContract.kt
package ru.sukharev.myrecipes.addrecipe

import ru.sukharev.myrecipes.base.BasePresenter
import ru.sukharev.myrecipes.base.BaseView

/**
 * Created by pasha on 13.11.17.
 */

interface AddRecipeContract {

    interface View : BaseView<Presenter> {

        fun onRecipeAdded()

        fun onRecipeAddingError()

        fun setRating(rating: Int)
    }

    interface Presenter : BasePresenter<View> {

        fun addRecipe(title: String, desc: String)

        fun setRating(rating: Int)
    }
}
